### PR TITLE
Detect multipart mime and ignore it

### PR DESCRIPTION
**Pull request by gabriel-samfira**
# Detect multipart mime user-data

This change makes ignition ignore user-data that is multipart mime.

## How to use

Deploy a VM and add multipart mime userdata. If the VM boots, it works.

## Testing done

Deployed a VM on OpenStack and set its userdata to a multipart mime userdata. The VM succeeded to boot.

- [x] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)
- [x] Inspected CI output for image differences: `/boot` and `/usr` size, packages, list files for any missing binaries, kernel modules, config files, kernel modules, etc.

Replaces: #823 

/update-sdk

### CODE REVIEW

**Review by pothos — COMMENTED** [review]
This needs a changelog entry and the commit IDs should point to merge commits, i.e., first merge the other PRs.

**Comment by gabriel-samfira** [comment]
> This needs a changelog entry and the commit IDs should point to merge commits, i.e., first merge the other PRs.

I will gladly change the commits and add the changelog once the coreos-cloudinit and mantle PRs merge :smile:. Just need some LGTMs on them.

File: changelog/changes/2023-06-08-cloudinit-multipart.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- Added support for multipart MIME userdata (scripts#873)
```
Comment (krnowak) [inline changelog/changes/2023-06-08-cloudinit-multipart.md:1]: Could you mention if this is for ignition or cloudconfig or both?

File: sdk_container/src/third_party/coreos-overlay/coreos-base/coreos-cloudinit/coreos-cloudinit-9999.ebuild
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/coreos-base/coreos-cloudinit/coreos-cloudinit-9999.ebuild:1]: Please bump the `coreos-cloudinit-1.14.0-r4.ebuild` symlink to `r5`.

File: sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0013-config-util-add-cloud-init-detection-to-initial-pars.patch
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0013-config-util-add-cloud-init-detection-to-initial-pars.patch:1]: Please bump the `ignition-2.15.0-r1.ebuild` symlink to `r2`.

**Comment by gabriel-samfira** [comment]
@krnowak changes made. Do you think it's worth running the test workflows again?

**Comment by krnowak** [comment]
> @krnowak changes made.

Cool, thanks!

> Do you think it's worth running the test workflows again?

Nah.

**Pull request merged**